ansible/roles/ocp-workload-fsi-cc-dispute-demo/tasks/workload.yml

```diff
@@ -0,0 +1,148 @@
+---
+- name: define ocp_project
+  set_fact:
+    ocp_project: "fsi-cc-dispute-{{guid}}"
+
+    # Templates come from here: https://raw.githubusercontent.com/jorgemoralespou/ose-sample-apps-layouts
+
+- name: Create project for FSI Credit Card Dispute demo
+  shell: |
+         oc new-project {{ocp_project}} \
+         --display-name="FSI Credit Card Dispute Demo" \
+         --description="FSI Credit Card Dispute Demo"
+  ignore_errors: true
```

Comment:
suggestion here instead of `ignore_errors: true`:
- have another task that test if the project exist under this user  (`changed_when: false`, `failed_when: ..TBD..`)
- create the project only if it doesn't already exist

Reply:
Could you give a proper example of this? I came up with the following, but not sure if this is what you're looking for:

```
- name: Check if project exists
  shell: |
         oc get project {{ocp_project}}
  changed_when: false
  register: project_exists_result
  ignore_errors: true

- name: Create project for FSI Credit Card Dispute demo
  shell: |
         oc new-project {{ocp_project}} \
         --display-name="FSI Credit Card Dispute Demo" \
         --description="FSI Credit Card Dispute Demo"
  when: project_exists_result is failed
```

But when doing this, if the project already exists, all sort of other errors will happen, e.g. ImageStreams that already exist, applications that already exist, etc. Which would mean that we would need to check every asset that we create through the playbook before we create it ...

WDYT?

Reply:
Why not use the k8s Ansible module instead of the oc command? Isn't it idempotent for you?